crates/router/src/connector/rapyd/transformers.rs

```diff
@@ -548,3 +616,74 @@ impl TryFrom<types::PaymentsCancelResponseRouterData<RapydPaymentsResponse>>
         })
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct RapydIncomingWebhook {
+    pub id: String,
+    #[serde(rename = "type")]
+    pub webhook_type: RapydWebhookObjectEventType,
+    pub data: WebhookData,
+    pub trigger_operation_id: Option<String>,
+    pub status: String,
+    pub created_at: i64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum RapydWebhookObjectEventType {
+    PaymentCompleted,
+    PaymentCaptured,
+    PaymentFailed,
+    RefundCompleted,
+    PaymentRefundRejected,
+    PaymentRefundFailed,
+}
+
+impl TryFrom<RapydWebhookObjectEventType> for api::IncomingWebhookEvent {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(value: RapydWebhookObjectEventType) -> Result<Self, Self::Error> {
+        match value {
+            RapydWebhookObjectEventType::PaymentCompleted => Ok(Self::PaymentIntentSuccess),
+            RapydWebhookObjectEventType::PaymentCaptured => Ok(Self::PaymentIntentSuccess),
+            RapydWebhookObjectEventType::PaymentFailed => Ok(Self::PaymentIntentFailure),
+            _ => Err(errors::ConnectorError::WebhookEventTypeNotFound).into_report()?,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+pub enum WebhookData {
+    PaymentData(ResponseData),
+    RefundData(RefundResponseData),
+}
+
+impl From<ResponseData> for RapydPaymentsResponse {
+    fn from(value: ResponseData) -> Self {
+        Self {
+            status: Status {
+                error_code: "".to_owned(),
+                status: "SUCCESS".to_owned(),
```

Comment:
why we are hard coding `SUCCESS`?

Status of connector can't be any string. There will be limited set of status right?

Reply:
i couldn't find it in there doc, this is from the sample implementation where its defined as string. anyway in case we get the data field this status comes as `SUCCESS` which mean the api was a success

Reply:
What is this status field, What if the api is success and status is failure?

Reply:
If the status is mentioned as string, we need to use their response status only, we can't override with a hardcoded status.

Reply:
we first check the status and update the payment status , there won't be a case in webhook where this hardcoded value is used to update status